skmultiflow/classification/trees/hoeffding_adaptive_tree.py

```diff
@@ -0,0 +1,489 @@
+                self.error_change = False
+
+            if (self.error_change is True):
+                self._alternate_tree = hat._new_learning_node()  # check call to new learning node
+                hat._alternateTrees += 1
+
+            elif (self._alternate_tree is not None and self._alternate_tree.is_null_error() is False):
+                if (self.get_error_width() > 300 and self._alternate_tree.get_error_width() > 300):
+                    old_error_rate = self.get_error_estimation()
+                    alt_error_rate = self._alternate_tree.get_error_estimation()
+                    fDelta = .05
+                    fN = 1.0 / self._alternate_tree.get_error_width() + 1.0 / (self.get_error_width())
+
+                    # CHECK
+                    bound = 1.0 / math.sqrt(
+                        2.0 * old_error_rate * (1.0 - old_error_rate) * math.log(2.0 / fDelta) * fN)
+                    # To check, bound never less than (old_error_rate - alt_error_rate)
+                    if bound < (old_error_rate - alt_error_rate):
+                        hat._active_leaf_node_cnt -= self.number_leaves()
+                        hat._active_leaf_node_cnt += self._alternate_tree.number_leaves()
+                        self.kill_tree_childs(hat)
+
+                        if parent is not None:
+                            parent.set_child(parent_branch, self._alternate_tree)
+                        else:
+                            hat._tree_root = hat._tree_root.alternateTree
+                        hat._switchAlternateTrees += 1
+                    elif (bound < alt_error_rate - old_error_rate):
+                        if isinstance(self._alternate_tree, HAT.ActiveLearningNode):
+                            self._alternate_tree = None
+                        elif (isinstance(self._alternate_tree, HAT.ActiveLearningNode)):
+                            self._alternate_tree = None
+                        else:
+                            self._alternate_tree.kill_tree_childs(hat)
+                        hat._prunedalternateTree += 1  # hat._pruned_alternate_trees to check
+
+            # Learn_From_Instance alternate Tree and Child nodes
+            if self._alternate_tree is not None:
+                self._alternate_tree.learn_from_instance(X, y, weight, hat, parent, parent_branch)
+
+            child_branch = self.instance_child_index(X)
+            child = self.get_child(child_branch)
+
+            if child is not None:
+                child.learn_from_instance(X, y, weight, hat, parent, parent_branch)
+
+        # Override NewNode
+        def kill_tree_childs(self, hat):
+            for child in self._children:
+                if child is not None:
+                    # Delete alternate tree if it exists
+                    # TODO (check the casting to object)
+                    # if (isinstance(child, HAT.AdaSplitNode) and HAT.NewNode(HAT.AdaSplitNode(child)) is not None):
```

Comment:
The condition should be:
``` python
if (isinstance(child, HAT.AdaSplitNode) and child._alternate_tree is not None)
```

```isinstance``` makes the casting redundant.

- [ ] Uncomment this section

Reply:
edited!

Reply:
Done!